### DISCUSSION TITLE
Source code for PRI-generation tasks?

### DISCUSSION THREAD

**Discussion by wjk** [microsoft/WindowsAppSDK#1444] [Q&A] [answered]
It was stated in https://github.com/microsoft/WindowsAppSDK/discussions/899#discussioncomment-823972 that while the PRI generation infrastructure used in the WinUI 3 package is not currently open-source (and there is no timeline for changing that), the version used standalone in MRT Core is. This, unfortunately, is misleading.

The _targets_ may be open-source and fully editable, but the _tasks_ used are not. If one is open-sourced, the other should be as well. Any changes made to the targets by the community must guess at the semantics of the tasks being called (and/or attempt to reverse-engineer their meaning via ILSpy), which is a very bad thing. The VS PRI generation targets are extremely obtuse and overcomplicated (not to mention that they requires all 9 GB of the VS UWP workload to be installed just to generate a PRI file). I can create, and fully control the creation of, a PRI file using exactly one target, as described in https://github.com/microsoft/WindowsAppSDK/issues/11#issuecomment-770476069. I still cannot follow the logic of the VS PRI generation targets in the slightest, which prevents me from customizing the generation behavior to differ from the “recommended” defaults.

-----

On a related note, I observed some time ago that the Reunion/Windows App SDK development process has been continuously characterized by poorly thought-out, half-assed attempts at open-sourcing and community involvement. The only once-closed components to have their source released to date are are bug-free, well-designed, and that no one needs to change (namely, MRT Core). (The other, “developed in the open” components are all too skeletal and difficult to integrate to be of any use; cf https://github.com/microsoft/WindowsAppSDK/discussions/987.) Since no one inside MS seems to have time to fix the myriad bugs in WinUI 3 (presumably because everyone is now too busy implementing ill-advised, publicly panned “simplifications” to the Windows 11 shell), no one else is able to fix the bugs either, since all the code that needs fixing remains internal-only. This observation has not changed in the slightest over the past year.

At this point my remaining goodwill for the Windows App SDK as a concept has just about run out. I still use Windows Forms for all my development because it is the only C# UI technology that Just Works™, but Windows Forms more-or-less forces the use of Windows 7-era UI and design patterns. If you really want developers to make their apps look modern, there must be an easy, simple, and (dare I say it) fun way to do so. If not, the current situation will become self-perpetuating (no one uses any of these technologies because they suck so badly, so no effort is given to improve them due to lack of use).

**Comment by axelandrejs** [comment, marked as answer]
The build targets are indeed an amalgamation of the SDK/VS tasks and our own custom targets. We would like to do better here, but we have no concrete plans yet. Sorry that is not the most satisfying answer. 

If you have specific questions around modifying the targets we may be able to help.
